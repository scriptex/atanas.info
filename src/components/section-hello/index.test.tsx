import * as React from 'react';
import { render } from '@testing-library/react';

import { mockUseEffect } from '../../scripts/shared';
import { Slider, SectionHello } from '.';

describe('Slider component', () => {
	it('Should render the Slider component', () => {
		const { asFragment, unmount } = render(<Slider />);

		expect(asFragment()).toMatchSnapshot();

		unmount();

		expect(asFragment()).toMatchSnapshot();
	});
});

describe('SectionHello component', () => {
	mockUseEffect();

	it('Should render the SectionHello component', () => {
		const { asFragment, unmount } = render(<SectionHello />);

		expect(asFragment()).toMatchSnapshot();

		unmount();

		expect(asFragment()).toMatchSnapshot();
	});
});
