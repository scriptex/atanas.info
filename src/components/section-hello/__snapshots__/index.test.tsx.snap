// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHello component Should render the SectionHello component 1`] = `
<Component
  hasButton={false}
  hasShell={true}
  id="hello"
  wrapperClassName="o-main--high"
>
  <div
    className="c-canvas"
    id="canvas"
  />
  <Component />
</Component>
`;

exports[`Slider component Should render the Slider component 1`] = `
<div
  className="c-slider"
>
  <h1>
    ðŸ‘‹ Hello! 
    <br />
    I am Atanas Atanasov
  </h1>
  <h2>
    <span
      className="current"
      key="0"
    >
      a lead front-end developer at E.ON
    </span>
    <span
      className=""
      key="1"
    >
      a senior software engineer at Toptal
    </span>
    <span
      className=""
      key="2"
    >
      a founding member at Braintrust
    </span>
    <span
      className=""
      key="3"
    >
      a TypeScript fanatic
    </span>
    <span
      className=""
      key="4"
    >
      a CSS wizard
    </span>
    <span
      className=""
      key="5"
    >
      a JavaScript tinkerer
    </span>
    <span
      className=""
      key="6"
    >
      a minimalist
    </span>
    <span
      className=""
      key="7"
    >
      an open source enthusiast
    </span>
    <span
      className=""
      key="8"
    >
      a maker of things
    </span>
    <span
      className=""
      key="9"
    >
      an a11y and UX advocate
    </span>
    <span
      className=""
      key="10"
    >
      a front-end magician
    </span>
    <span
      className=""
      key="11"
    >
      a bug fixer
    </span>
    <span
      className=""
      key="12"
    >
      a senior web developer
    </span>
    <span
      className=""
      key="13"
    >
      I am into data visualization
    </span>
    <span
      className=""
      key="14"
    >
      an UI architect
    </span>
    <span
      className=""
      key="15"
    >
      I am obsessed with quality
    </span>
    <span
      className=""
      key="16"
    >
      a web standards fanatic
    </span>
    <span
      className=""
      key="17"
    >
      a JavaScript master
    </span>
    <span
      className=""
      key="18"
    >
      a NodeJS hacker
    </span>
    <span
      className=""
      key="19"
    >
      a father
    </span>
    <span
      className=""
      key="20"
    >
      a husband
    </span>
  </h2>
</div>
`;
